docker/theia-app-theia-ext/Dockerfile

```diff
@@ -0,0 +1,44 @@
+FROM node:18.20.4-bookworm-slim AS build
+ARG APP_SUBFOLDER
+
+RUN apt-get update && apt-get install -y \
+    git \
+    python3 \
+    make \
+    pkg-config \
+    libsecret-1-dev \
+    g++ \
+    libx11-dev \
+    libxkbfile-dev
+
+# Build the browser theia-trace-extension application
+COPY ${APP_SUBFOLDER}/package.json /app/tte/
+COPY ${APP_SUBFOLDER}/webpack.config.js /app/tte/
+COPY yarn.lock  /app/tte/
```

Comment:
This line causes an error for me:
ERROR [build 5/7] COPY yarn.lock

Reply:
I have refactored the layout of the Docker example vs their theia applications and other resources. The Dockerfile(s ) are now in the "docker root" (\<repo\>/docker) and so I think it will be much clearer that the images should be built from that folder. 

I have also much improved the READMEs.